eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/RatifySpec.hs

```diff
@@ -35,6 +35,117 @@ spec =
     votingSpec
     delayingActionsSpec
     spoVotesForHardForkInitiation
+    paramChangeAffectsProposalsSpec
+
+paramChangeAffectsProposalsSpec ::
+  forall era.
+  ConwayEraImp era =>
+  SpecWith (ImpTestState era)
+paramChangeAffectsProposalsSpec =
+  describe "ParameterChange affects existing proposals" $ do
+    let largerThreshold = 51 %! 100
+        smallerThreshold = 1 %! 2
+    describe "DRep" $ do
+      let setThreshold threshold = do
+            drepVotingThresholds <- getsPParams ppDRepVotingThresholdsL
+            modifyPParams $
+              ppDRepVotingThresholdsL
+                .~ (drepVotingThresholds & dvtCommitteeNormalL .~ threshold)
+          enactThreshold threshold drepC hotCommitteeC = do
+            drepVotingThresholds <- getsPParams ppDRepVotingThresholdsL
+            let paramChange =
+                  ParameterChange
+                    SNothing
+                    ( emptyPParamsUpdate
+                        & ppuDRepVotingThresholdsL
+                          .~ SJust (drepVotingThresholds & dvtCommitteeNormalL .~ threshold)
+                    )
+                    SNothing
+            pcGai <- submitGovAction paramChange
+            submitYesVote_ (DRepVoter drepC) pcGai
+            submitYesVote_ (CommitteeVoter hotCommitteeC) pcGai
+            passNEpochs 2
+          submitTwoExampleProposalsAndVoteOnTheChild gpiCC drepKH = do
+            committeeC <- KeyHashObj <$> freshKeyHash
+            let updateCC parent = UpdateCommittee parent mempty (Map.singleton committeeC $ EpochNo 5) $ 1 %! 2
+            gai' <- submitGovAction $ updateCC $ SJust gpiCC
+            -- We submit a descendent proposal so that even though it is sufficiently
+            -- voted on, it cannot be ratified before the ParameterChange proposal
+            -- is enacted.
+            gai <- submitGovAction $ updateCC $ SJust $ GovPurposeId gai'
+            submitYesVote_ (DRepVoter $ KeyHashObj drepKH) gai
```

Comment:
```suggestion
            gaiParent <- submitGovAction $ updateCC $ SJust gpiCC
            -- We submit a descendent proposal so that even though it is sufficiently
            -- voted on, it cannot be ratified before the ParameterChange proposal
            -- is enacted.
            gaiChild <- submitGovAction $ updateCC $ SJust $ GovPurposeId gaiParent
            submitYesVote_ (DRepVoter $ KeyHashObj drepKH) gaiChild
```